pallets/order-book/src/tests.rs

```diff
@@ -380,6 +389,325 @@ fn fill_order_full_works() {
+					RuntimeEvent::OrderBook(Event::OrderUpdated {
+						order_id,
+						account: order.placing_account,
+						buy_amount: remaining_buy_amount,
+						sell_rate_limit: order.max_sell_rate,
+						min_fulfillment_amount: order.min_fulfillment_amount,
+					})
+				);
+				assert_eq!(
+					System::events()[4].event,
+					RuntimeEvent::OrmlTokens(orml_tokens::Event::Transfer {
+						currency_id: DEV_AUSD_CURRENCY_ID,
+						to: ACCOUNT_0,
+						from: ACCOUNT_1,
+						amount: partial_buy_amount
+					})
+				);
+				assert_eq!(
+					System::events()[5].event,
+					RuntimeEvent::OrmlTokens(orml_tokens::Event::Transfer {
+						currency_id: DEV_USDT_CURRENCY_ID,
+						to: ACCOUNT_1,
+						from: ACCOUNT_0,
+						amount: expected_sell_amount
+					})
+				);
+				assert_eq!(
+					System::events()[6].event,
+					RuntimeEvent::OrderBook(Event::OrderFulfillment {
+						order_id,
+						placing_account: order.placing_account,
+						fulfilling_account: ACCOUNT_1,
+						partial_fulfillment: true,
+						fulfillment_amount: partial_buy_amount,
+						currency_in: order.asset_in_id,
+						currency_out: order.asset_out_id,
+						sell_rate_limit: order.max_sell_rate,
+					})
+				);
+			});
+		}
+	}
+
+	#[test]
+	fn fill_order_partial_with_full_amount_works() {
+		new_test_ext().execute_with(|| {
+			let buy_amount = 100 * CURRENCY_AUSD_DECIMALS;
+			let min_fulfillment_amount = 1 * CURRENCY_AUSD_DECIMALS;
+			let sell_ratio = FixedU128::checked_from_rational(3u32, 2u32).unwrap();
+
+			assert_ok!(OrderBook::place_order(
+				ACCOUNT_0,
+				DEV_AUSD_CURRENCY_ID,
+				DEV_USDT_CURRENCY_ID,
+				buy_amount,
+				sell_ratio,
+				min_fulfillment_amount,
+			));
+
+			let (order_id, order) = get_account_orders(ACCOUNT_0).unwrap()[0];
+
+			assert_ok!(OrderBook::fill_order_partial(
+				RuntimeOrigin::signed(ACCOUNT_1),
+				order_id,
+				buy_amount,
+			));
+
+			assert_eq!(
+				AssetPairOrders::<Runtime>::get(DEV_AUSD_CURRENCY_ID, DEV_USDT_CURRENCY_ID),
+				vec![]
+			);
+
+			let max_sell_amount = OrderBook::convert_with_ratio(
+				order.asset_in_id,
+				order.asset_out_id,
+				order.max_sell_rate,
+				buy_amount,
+			)
+			.unwrap();
+
+			assert_eq!(
+				System::events()[2].event,
+				RuntimeEvent::OrmlTokens(orml_tokens::Event::Unreserved {
+					currency_id: DEV_USDT_CURRENCY_ID,
+					who: ACCOUNT_0,
+					amount: max_sell_amount
+				})
+			);
+			assert_eq!(
+				System::events()[3].event,
+				RuntimeEvent::OrmlTokens(orml_tokens::Event::Transfer {
+					currency_id: DEV_AUSD_CURRENCY_ID,
+					to: ACCOUNT_0,
+					from: ACCOUNT_1,
+					amount: buy_amount
+				})
+			);
+			assert_eq!(
+				System::events()[4].event,
+				RuntimeEvent::OrmlTokens(orml_tokens::Event::Transfer {
+					currency_id: DEV_USDT_CURRENCY_ID,
+					to: ACCOUNT_1,
+					from: ACCOUNT_0,
+					amount: max_sell_amount
+				})
+			);
+			assert_eq!(
+				System::events()[5].event,
+				RuntimeEvent::OrderBook(Event::OrderFulfillment {
```

Comment:
Nit: Could we also ensure the order was cleared from storage? Checking for the event does this implicitly but since clearing an order and dispatching the _full_ fulfillment event are decoupled, we need to check the former as well.

Reply:
Sure, adding that check now. Thank you!